# Datasets

This is a growing repository of datasets broadly related to culture and the humanities. The sources of the datasets as well as brief descriptions and example uses can be found below.

If you'd like to add a dataset or an example use case to this page, please open [an issue on GitHub](https://github.com/melaniewalsh/Intro-Cultural-Analytics/issues) or email me at melanie.walsh@cornell.edu

## Film 🎬


### Hollywood Film Dialogue By Character Gender and Age
(1925-2015)

Get the data: {download}`Download Hollywood Film Dialogue data <../data/Pudding/Pudding-Film-Dialogue-Clean.csv>`  
Original source: Hannah Anderson and Matt Daniels, *The Pudding*

Brief description:  

This {download}`CSV file <../data/Pudding/Pudding-Film-Dialogue-Clean.csv>` is a consolidated and slightly modified version of data shared by Hannah Anderson and Matt Daniels for their *Pudding* article, ["Film Dialogue from 2,000 screenplays, Broken Down by Gender and Age"](https://pudding.cool/2017/03/film-dialogue/). The original datasets can be found [on GitHub here](https://github.com/matthewfdaniels/scripts/). 

For 2,000 films from 1925 to 2015, the dataset includes information about characters' names, genders, ages, how many words they spoke in each film, as well as the release year of each film and how much money the film grossed. Anderson and Daniels determined character age and gender (which they code as binary) based on corresponding IMDB  information for actors. They acknowledge this is an imperfect approach. For more on the compilation of the dataset and their methodology, see [FAQ for the “Film Dialogue, By Gender” Project](https://medium.com/@matthew_daniels/faq-for-the-film-dialogue-by-gender-project-40078209f751).

Example uses:
- ["Film Dialogue from 2,000 screenplays, Broken Down by Gender and Age"](https://pudding.cool/2017/03/film-dialogue/), *The Pudding*
- ["Pandas Basics Part 3"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Data-Analysis/Pandas-Basics-Part3.html), *Introduction to Cultural Analytics & Python*

Preview dataset

In [1]:
import pandas as pd
pd.set_option("max_rows", 150)
pd.read_csv("../data/Pudding/Pudding-Film-Dialogue-Clean.csv").sample(10)

title  release_year  \
19885                                        Noah          2014   
15107                      Star Trek: Generations          1994   
21000  Star Wars: Episode VII - The Force Awakens          2015   
600                                  Broken Arrow          1996   
21607                              Brick Mansions          2014   
10975                        Terminator Salvation          2009   
20247                  Die Hard: With a Vengeance          1995   
10402                                      S1m0ne          2002   
3600                   10 Things I Hate About You          1999   
9879                               Reindeer Games          2000   

             character gender  words  proportion_of_dialogue   age  gross  \
19885             Shem    man    547                0.089997  22.0  107.0   
15107            Lursa  woman    130                0.014183   NaN  157.0   
21000            C-3Po    man    138                0.020702  69.0  927.0   
600    Secretary Of De    man    291                0.034809  53.0  138.0   
21607  Grandfather Col    man    182                0.021237  78.0   21.0   
10975         Morrison    man    133                0.032102   NaN  145.0   
20247     Simon Gruber    man   1185                0.178814  47.0  200.0   
10402  Viktor Taransky    man  10520                0.408766  62.0   14.0   
3600     Kat Stratford  woman   4718                0.239736  18.0   65.0   
9879             Jumpy    man    964                0.042140  56.0   37.0   

       script_id  
19885       7659  
15107       5035  
21000       8099  
600          779  
21607       8410  
10975       3516  
20247       7818  
10402       3345  
3600        1512  
9879        3190

---

## Literature 📚

### *Lost in the City* (1993), Edward P. Jones
HathiTrust Extracted Features

Get the data: {download}`Download Lost in the City data <../texts/literature/Lost-in-the-City-HTRC-Extracted-Features.zip>`   
Original source: [HathiTrust Digital Library](https://wiki.htrc.illinois.edu/pages/viewpage.action?pageId=79069329)

Brief description:  

The {download}`Lost in the City HTRC Extracted Features zip file <../texts/literature/Lost-in-the-City-HTRC-Extracted-Features.zip>` contains word frequencies per page—or "extracted features"—made available by the [HathiTrust Digital Library](https://wiki.htrc.illinois.edu/pages/viewpage.action?pageId=79069329) for Edward P. Jones's short story collection *Lost in the City*. I have also added each short story title for the correct corresponding pages to the dataset.

There are three CSV files in the zip file. "Lost-in-the-City-HTRC-Extracted-Features(PerPage).csv" contains lowercased word frequencies per page as well as part-of-speech information (49,330 rows). "Lost-in-the-City-HTRC-Extracted-Features(PerStory).csv" contains lowercased word frequencies per story as well as part-of-speech information (20,748 rows). "Lost-in-the-City-HTRC-Extracted-Features(5PerStory).csv" contains pre-processed lowercased word frequencies (with stopwords and punctuation removed) for words that appear more than 5 times in a story as well as part-of-speech information (1,364 rows).

Example uses:
- ["TF-IDF With HathiTrust Data"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Text-Analysis/TF-IDF-HathiTrust.html), *Introduction to Cultural Analytics & Python*

Preview dataset

In [78]:
pd.read_csv("../texts/literature/Lost-in-the-City-HTRC-Extracted-Features/Lost-in-the-City-HTRC-Extracted-Features(PerStory).csv").sample(10)

story lowercase  pos  count
5836                         04: Young Lions      stay   VB      2
20317                            Back Matter  poignant   JJ      1
15767                             11: Gospel       own   JJ      2
16088                             11: Gospel     stool   NN      1
10747  07: The Sunday Following Mother's Day    seeing  VBG      1
6441                           05: The Store      born  VBN      1
6310                           05: The Store    around   IN     14
13443                 09: His Mother's House     outta   RB      1
16317                             11: Gospel   windows  NNS      1
18492                       13: A Dark Night     quiet   JJ      5

___

### African American Literature
(1853-1923)

Get the data: {download}`Download African American Literature data <../texts/literature/African-American-Literature-1853-1923.zip>`  
Source: Amardeep Singh

Brief description:  

The {download}`African American Literature (1853-1923) zip file <../texts/literature/African-American-Literature-1853-1923.zip>` contains 100 works of fiction and poetry by African American writers between 1853-1923. It also contains an Excel file of metadata about the publisher, publication data, and publication place of each text. This corpus was compiled and shared by Amardeep Singh. You can read more about this corpus and its creation in [Singh's blog post about the African American Literature corpus](http://www.electrostani.com/2020/07/announcing-open-access-african-american.html).

Lastly, Singh asks users of this data to adhere to the spirit of the [Colored Conventions Project's principles](https://coloredconventions.org/about-records/ccp-corpus/).

Preview dataset

In [57]:
pd.read_excel("../texts/literature/African American Literature Text Corpus/African American Literature Corpus Metadata-Amardeep Singh.xlsx")

Author (last, first)  \
0                                       Adams, Clayton   
1              Anderson, William and Walter H. Stowers   
2                                        Andrews, W.T.   
3                                    Ashby, William M.   
4                                        Bennett, John   
5                                      Bibb, Eloise A.   
6                                 Blackson, Lorenzo D.   
7                         Braithwaite, William Stanley   
8                         Braithwaite, William Stanley   
9                                 Brown, William Wells   
10                                  Bruce, John Edward   
11                               Burgess, Marie Louise   
12                             Bush-Banks, Olivia Ward   
13                             Bush-Banks, Olivia Ward   
14                                   Chesnutt, Charles   
15                                   Chesnutt, Charles   
16                                   Chesnutt, Charles   
17                                   Chesnutt, Charles   
18                                   Chesnutt, Charles   
19                           Clifford, Carrie Williams   
20                           Clifford, Carrie Williams   
21                              Coleman, Lucretia H.N.   
22                                   Cotter, Joseph S.   
23   Dandridge, Raymond GarfieThe Poet and Other Poems   
24   Dandridge, Raymond GarfieThe Poet and Other Poems   
25                                      Detter, Thomas   
26                                 Douglass, Frederick   
27                                   Downing, Henry F.   
28                                       Dreer, Herman   
29                                     Du Bois, W.E.B.   
30                                Dunbar-Nelson, Alice   
31                                Dunbar-Nelson, Alice   
32                               Dunbar, Paul Laurence   
33                               Dunbar, Paul Laurence   
34                               Dunbar, Paul Laurence   
35                               Dunbar, Paul Laurence   
36                               Dunbar, Paul Laurence   
37                               Dunbar, Paul Laurence   
38                               Durham, John Stephens   
39                            Ellis, George Washington   
40                            Fleming, Sarah Lee Brown   
41                            Fleming, Sarah Lee Brown   
42                                Fordham, Mary Weston   
43                                  Fowler, Charles H.   
44                             Fortune, Timothy Thomas   
45                              Franklin, James Thomas   
46                                   Fullilove, Maggie   
47                                Fulton, David Bryant   
48                                   Gilmore, F. Grant   
49                                   Gilmore, James R.   
50                                  Grant, John Wesley   
51                                   Griggs, Sutton E.   
52                                   Griggs, Sutton E.   
53                                   Griggs, Sutton E.   
54                                   Griggs, Sutton E.   
55                                   Griggs, Sutton E.   
56                               Harper, Frances, E.W.   
57                               Harper, Frances, E.W.   
58                               Harper, Frances, E.W.   
59                               Harper, Frances, E.W.   
60                               Harper, Frances, E.W.   
61                                    Heard, Josephine   
62                           Henderson, Elliott Blaine   
63                           Henderson, Elliott Blaine   
64                                    Hopkins, Pauline   
65                                    Hopkins, Pauline   
66                                    Hopkins, Pauline   
67                                    Hopkins, Pauline   
68                              

---

### Colonial South Asian Literature
(1850-1923)

Get the data: {download}`Download Colonial South Asian Literature data <../texts/literature/Colonial-South-Asian-Literature-1850-1923.zip>`  
Source: Amardeep Singh

Brief description:  

The {download}`Colonial South Asian Literature (1850-1923) zip file <../texts/literature/Colonial-South-Asian-Literature-1850-1923.zip>` contains ~100 works of literature by South Asian and British writers between 1853-1923. It also contains an Excel file of metadata about publication and the nationality of each other. This corpus was compiled and shared by Amardeep Singh. You can read more about this corpus and its creation in [Singh's blog post about the Colonial South Asian Literature corpus](http://www.electrostani.com/2020/08/text-corpus-colonial-south-asian.html).

Preview dataset

In [58]:
pd.read_excel("../texts/literature/Colonial South Asian Literature, 1850-1923/Colonial South Asian Literature Corpus.xlsx")

Author  \
0                                      Arnold, W.D.   
1                                        Bain, F.W.   
2                                    Banerjea, S.B.   
3                                   Candler, Edmund   
4                                   Candler, Edmund   
5                                   Candler, Edmund   
6                                   Candler, Edmund   
7                                   Candler, Edmund   
8                         Chatterji, Bankim Chandra   
9                         Chatterji, Bankim Chandra   
10                        Chatterji, Bankim Chandra   
11                        Chatterji, Bankim Chandra   
12                        Chatterji, Bankim Chandra   
13                          Chatterji, Saratchandra   
14                                    Croker, B.M.    
15                                    Croker, B.M.    
16                                    Croker, B.M.    
17                                      Diver, Maud   
18                                      Diver, Maud   
19                                      Diver, Maud   
20                                      Diver, Maud   
21                                      Diver, Maud   
22                      Derozio, Henry Louis Vivian   
23                         Dracott, Alice Elizabeth   
24                           Duncan, Sara Jeannette   
25                           Duncan, Sara Jeannette   
26                           Duncan, Sara Jeannette   
27                           Duncan, Sara Jeannette   
28                           Duncan, Sara Jeannette   
29                         Dutt, Michael Madhusudan   
30                                       Dutt, Toru   
31                                   Dyer, Helen S.   
32                                   Edwardes, S.M.   
33                                 Ellis, Edward S.   
34                               Faulkner, Georgene   
35                            Fenn, George Manville   
36                                   Forster, E. M.   
37                                        Frere, M.   
38                             Gibbon, Frederick P.   
39                                      Henty, G.A.   
40                                      Henty, G.A.   
41                                      Henty, G.A.   
42                                      Henty, G.A.   
43                                      Henty, G.A.   
44                              Hodson, Major W.S.R   
45                          Hossain, Rokeya Sakhwat   
46                                      Jacob, P.W.   
47         Khan, Inayat and Jessie Duncan Westbrook   
48                           Kipling, John Lockwood   
49                                 Kipling, Rudyard   
50                                 Kipling, Rudyard   
51                                 Kipling, Rudyard   
52                                 Kipling, Rudyard   
53                                 Kipling, Rudyard   
54                                 Kipling, Rudyard   
55                                 Kipling, Rudyard   
56                                 Kipling, Rudyard   
57                                 Kipling, Rudyard   
58                                 Kipling, Rudyard   
59                                 Kipling, Rudyard   
60           Kipling, Rudyard and Wolcott Balestier   
61                                  Mayo, Katherine   
62   McNair, Major J.F.A. and Thomas Lamberg Barlow   
63                              Mukerji, Dhan Gopal   
64                              Mukerji, Dhan Gopal   
65                              Mukerji, Dhan Gopal   
66                              Mukerji, Dhan Gopal   
67                              Mukerji, Dhan Gopal   
68                                    Mundy, Talbot   
69                                    Mundy, Talbot   
70                                    Mundy, Talbot   
71                                    Mundy, Talbot   
72                            

___

### txtLAB's Multilingual Novels
(1771-1932)

Get the data: [Link to Multilingual Novels data](https://figshare.com/articles/txtlab_Novel450/2062002/3)   
Source: Andrew Piper, McGill [txtLAB](https://txtlab.org/data-sets/)

Brief description:  

The [txtLAB's multilingual novels](https://figshare.com/articles/txtlab_Novel450/2062002/3) takes you to a repository where you can download a directory of 150 English-language novels, 150 German-language novels, and 150 French-language novels, which span from 1771 to 1932. Authors featured include Goethe, Franz Kafka, Hermann Melville, Mary Shelley, Kate Chopin, Virginia Woolf, Victor Hugo, Alexandre Dumas, and many more. These text files were compiled and shared by Andrew Piper and the txtLab.

Example uses:
- ["Novel Devotions: Conversional Reading, Computational Modeling, and the Modern Novel,"](http://piperlab.mcgill.ca/pdfs/Piper_NovelConversions.pdf) Andrew Piper

Preview dataset

In [51]:
print(open("../texts/literature/McGill_txtalb_Novel450/English/EN_1925_Woolf,Virginia_Mrs.Dalloway_Novel.txt").read())

MRS. DALLOWAY

 

by

 

Virginia Woolf

 

 

 

Mrs. Dalloway said she would buy the flowers herself.

For Lucy had her work cut out for her. The doors would be taken off their hinges; Rumpelmayer's men were coming. And then, thought Clarissa Dalloway, what a morning--fresh as if issued to children on a beach.

What a lark! What a plunge! For so it had always seemed to her, when, with a little squeak of the hinges, which she could hear now, she had burst open the French windows and plunged at Bourton into the open air. How fresh, how calm, stiller than this of course, the air was in the early morning; like the flap of a wave; the kiss of a wave; chill and sharp and yet (for a girl of eighteen as she then was) solemn, feeling as she did, standing there at the open window, that something awful was about to happen; looking at the flowers, at the trees with the smoke winding off them and the rooks rising, falling; standing and looking until Peter Walsh said, "Musing among the vegetables?

---

### Modernist Journal Data
(1890s-1920s)

Get the data: [Link to Modernist Journal data](https://sourceforge.net/projects/mjplab/files/)   
Source: [The Modernist Journals Project](https://modjourn.org/)

Brief description:  

The [Modernist journal data link](https://sourceforge.net/projects/mjplab/files/) takes you to a repository where you can download publication metadata for 14 modernist journals from the 1890s to the 1920s — such as *Poetry Magazine*, *The Little Review*, and *The Crisis*. The Modernist Journals Project, which has digitized these journals, provides CSV and tab-separated text files that contain information for every contributor and every work published in the journals.  

Example uses:
- [Comparative Charts―Involving Contributions to The Egoist, The Little Review, and Others (1915-1919)](https://modjourn.org/comparative-charts%e2%80%95involving-contributions-to-the-egoist-the-little-review-and-others-1915-1919/), The Modernist Journals Project

Preview dataset

In [20]:
pd.read_csv("../data/Crisis_3.everycontributor.txt", delimiter="|").sample(5)

contributor                               title   \
268               Clarana, José          The Schooling of the Negro    
1415   Johnson, Georgia Douglas    A Sonnet in Memory of John Brown    
964        Jackson, Virginia P.                              Africa    
1208  Jordan, Winifred Virginia                              Values    
639            Stevenson, Helen                  How I Grew My Corn    

          genre      pages    volume    issue          date   journal title   \
268    articles    133-136          6        3   1913-07-01          Crisis    
1415     poetry    169-169         24        4   1922-08-01          Crisis    
964      poetry    166-166         17        4   1919-02-01          Crisis    
1208     poetry      15-15         21        1   1920-11-01          Crisis    
639    articles    273-274         12        6   1916-10-01          Crisis    

                   journal subtitle               issue name   \
268    A Record of the Darker Races       Educational Number    
1415   A Record of the Darker Races                             
964    A Record of the Darker Races    Reconstruction Number    
1208   A Record of the Darker Races                             
639    A Record of the Darker Races      [Children's Number]    

                 journal editor   \
268    Du Bois, W. E. Burghardt    
1415   Du Bois, W. E. Burghardt    
964    Du Bois, W. E. Burghardt    
1208   Du Bois, W. E. Burghardt    
639    Du Bois, W. E. Burghardt    

                                             publisher   journal location   \
268    National Association for the Advancement of C...          New York    
1415   National Association for the Advancement of C...          New York    
964    National Association for the Advancement of C...          New York    
1208   National Association for the Advancement of C...          New York    
639    National Association for the Advancement of C...          New York    

       issue length (pp)    issue height (cm)    issue width (cm)  
268                    52                 22.9               15.2  
1415                   48                 22.9               15.2  
964                    52                 22.9               15.2  
1208                   52                 22.9               15.2  
639                    52                 22.9               15.2

---

### Seattle Public Library Circulation Data
(2005-present)

Get the data: {download}`Download Seattle Public Library Circulation Data (Books checked out > 10 times between 2010-2017 <../data/Seattle_Book_Checkouts_2010_2017.zip>` // [Link to All Seattle Public Library Circulation Data (2005-present)](https://data.seattle.gov/Community/Checkouts-by-Title/tmmm-ytt6/data)  
Source: [City of Seattle](http://www.seattle.gov/tech/initiatives/open-data/about-the-open-data-program)

Brief description:  
The [Seattle Public Library check-out data link](https://data.seattle.gov/Community/Checkouts-by-Title/tmmm-ytt6/data) takes you to a database that contains circulation data about the Seattle Public Library system from 2005 until the present. You can filter this database or search for keywords (e.g., "James Baldwin") and export a file of the filtered data by clicking "Export" and your desired file type.

Example uses:
- ["James Baldwin Checkouts from the SPL,"](http://tweetsofanativeson.com/Seattle-Public-Library/) Melanie Walsh

Preview dataset

In [21]:
pd.read_csv("../data/Seattle_Book_Checkouts_2010_2017.csv").sample(5)

UsageClass CheckoutType MaterialType  CheckoutYear  CheckoutMonth  \
259350   Physical      Horizon         BOOK          2012              3   
580809   Physical      Horizon         BOOK          2014              2   
116767   Physical      Horizon         BOOK          2010              3   
241495   Physical      Horizon         BOOK          2015              8   
548316   Physical      Horizon         BOOK          2014              3   

        Checkouts                                              Title  \
259350         14  The growing story / by Ruth Krauss ; illustrat...   
580809         10             Frommers easyguide to New Orleans 2014   
116767         32           The privileges : a novel / Jonathan Dee.   
241495         12  ABC school's for me! / written by Susan B. Kat...   
548316        100                     Uncataloged Folder or Bag--GWD   

                      Creator  \
259350           Krauss, Ruth   
580809                    NaN   
116767          Dee, Jonathan   
241495  Katz, Susan B., 1971-   
548316                    NaN   

                                                 Subjects          Publisher  \
259350  Seasons Juvenile fiction, Growth Juvenile fiction     HarperCollins,   
580809                          New Orleans La Guidebooks                NaN   
116767  Rich people Fiction, Families Fiction, New Yor...      Random House,   
241495  Alphabet Juvenile fiction, Schools Juvenile fi...  Scholastic Press,   
548316                                                NaN                NaN   

       PublicationYear  
259350            2007  
580809             NaN  
116767            2010  
241495          [2015]  
548316             NaN

---

### Game of Thrones Character Relationships

Get the data: {download}`Download Game of Thrones Character data <../data/game-of-thrones-characters.zip>`  
Source: A. Beveridge and J. Shan

Brief description:  

The {download}`Game of Thrones Character Relationships zip file <../data/game-of-thrones-characters.zip>`  contains network data for character relationships within George R. R. Martin's *A Storm of Swords*, the third novel in his series *A Song of Ice and Fire* (also known as the HBO television adaptation *Game of Thrones*). This data was originally compiled by A. Beveridge and J. Shan for their article, ["Network of Thrones"](https://www.maa.org/sites/default/files/pdf/Mathhorizons/NetworkofThrones%20%281%29.pdf).

The nodes csv contains 107 different characters, and the edges csv contains 353 weighted relationships between those characters, which were calculated based on how many times two characters' names appeared within 15 words of one another in the novel. For more on the methodology, see Beveridge and Shan's [original article](https://www.maa.org/sites/default/files/pdf/Mathhorizons/NetworkofThrones%20%281%29.pdf).

Example uses:
- ["Network of Thrones,"](https://www.maa.org/sites/default/files/pdf/Mathhorizons/NetworkofThrones%20%281%29.pdf), A. Beveridge and J. Shan
- ["Network Analysis,"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Network-Analysis/Network-Analysis.html) *Introduction to Cultural Analytics & Python*

Preview dataset

In [60]:
pd.read_csv("../data/got-edges.csv").sample(10)

Source     Target  Weight
15           Arya      Jaime      11
154           Jon      Aemon      30
150       Joffrey     Tommen       9
170           Jon    Stannis       9
152         Jojen      Meera      33
140       Joffrey     Gregor       5
241          Robb      Theon      11
250  Robert Arryn  Marillion       4
2           Aerys      Jaime      18
274         Sansa      Loras      14

---

## Politics 🗳️ & History 📜


### *The New York Times* Obituaries
(1852-2007)

Get the data: {download}`Download *New York Times* Obituaries data <../texts/history/NYT-Obituaries.zip>`     
Original source: [Matthew Lavin](https://programminghistorian.org/en/lessons/analyzing-documents-with-tfidf#lesson-dataset)

Brief description:  

The {download}`*New York Times* obituaries zip file (.zip) of text files (.txt) <../texts/history/NYT-Obituaries.zip>` contains 379 *New York Times* obituaries (1852-2007) based on those collected by Matt Lavin for his *Programming Historian* tutorial, [Analyzing Documents with TF-IDF](https://programminghistorian.org/en/lessons/analyzing-documents-with-tfidf#lesson-dataset).

I re-scraped the 366 obituaries included in Lavin's tutorial so that the obituary subject's name and death year is included in each text file name. I also added 13 more ["Overlooked"](https://www.nytimes.com/interactive/2018/obituaries/overlooked.html) obituaries — belated obituaries of remarkable women and minorities who did not receive a *NYT* obituary at the time of their death. Obituary subjects include academics, military generals, artists, athletes, activists, politicians, and businesspeople — such as Ada Lovelace, Ulysses Grant, Marilyn Monroe, Virginia Woolf, Jackie Robinson, Marsha P. Johnson, Cesar Chavez, John F. Kennedy, Ray Kroc, and many more.

Example uses:
- [Analyzing Documents with TF-IDF](https://programminghistorian.org/en/lessons/analyzing-documents-with-tfidf), Matt Lavin
- ["Topic Modeling Text Files"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Text-Analysis/Topic-Modeling-Text-Files.html), *Introduction to Cultural Analytics & Python*

Preview dataset

In [26]:
print(open("../texts/history/NYT-Obituaries/1852-Ada-Lovelace.txt").read())

A gifted mathematician who is now recognized as the first computer programmer.By CLAIRE CAIN MILLER

 A century before the dawn of the computer age, Ada Lovelace imagined the modern-day, general-purpose computer. It could be programmed to follow instructions, she wrote in 1843. It could not just calculate but also create, as it “weaves algebraic patterns just as the Jacquard loom weaves flowers and leaves.” The computer she was writing about, the British inventor Charles Babbage’s Analytical Engine, was never built. But her writings about computing have earned Lovelace — who died of uterine cancer in 1852 at 36 — recognition as the first computer programmer. 

 The program she wrote for the Analytical Engine was to calculate the seventh Bernoulli number. (Bernoulli numbers, named after the Swiss mathematician Jacob Bernoulli, are used in many different areas of mathematics.) But her deeper influence was to see the potential of computing. The machines could go beyond calculating numbers

---
        
### U.S. Inaugural Addresses
(1789-2017)

Get the data: {download}`Download U.S. Inaugural Addresses data <../texts/history/US_Inaugural_Addresses.zip>`

Brief description: 

The {download}`U.S. Inaugural Addresses zip file (.zip) of text files (.txt) <../texts/history/US_Inaugural_Addresses.zip>` contains U.S. Inaugural Addresses ranging from President George Washington (1789) to President Donald Trump (2017). Each text file is titled with a number, the corresponding last name of the U.S. President, and the corresponding year of the Inaugural Address.

Example uses:
- ["jsLDA: In-browser topic modeling,"](https://github.com/mimno/jsLDA) David Mimno

Preview dataset

In [65]:
print(open("../texts/history/US_Inaugural_Addresses/56_obama_2009.txt").read())

Barack Obama	1/20/2009	My fellow citizens, I stand here today humbled by the task before us, grateful for the trust you have bestowed, mindful of the sacrifices borne by our ancestors. I thank President Bush for his service to our Nation, as well as the generosity and cooperation he has shown throughout this transition.  Forty-four Americans have now taken the Presidential oath. The words have been spoken during rising tides of prosperity and the still waters of peace. Yet every so often, the oath is taken amidst gathering clouds and raging storms. At these moments, America has carried on not simply because of the skill or vision of those in high office, but because we the people have remained faithful to the ideals of our forebears and true to our founding documents.  So it has been; so it must be with this generation of Americans.  That we are in the midst of crisis is now well understood. Our Nation is at war against a far-reaching network of violence and hatred. Our economy is badl

---

### Nobel Prize Winners
(1901-2017)

Get the data: {download}`Download Nobel Prize Winners data <../data/nobel-prize-winners.zip>`  
Original source: The European Data Portal and [the official Nobel Prize API](https://www.nobelprize.org/about/developer-zone-2/)

Brief description:  

The {download}`Nobel Prize winners CSV file (.CSV) <../data/nobel-prize-winners.zip>` contains information about 957 Nobel Prize winners from 1901 to 2017. This information includes the Nobel laureate's name, birth and death date (if applicable), birth and death location (plus **latitude and longitude coordinates** for the locations), the year they won the Nobel Prize, the category of the Nobel Prize, and the "motivation" for the Nobel Prize.

Nobel laureates include Marie Curie, Johannes Stark, Woodrow Wilson, Jane Addams, Rabindranath Tagore, John Steinbeck, Gabriel Garcia Marquez, Karl Ziegler, Toni Morrison, and many more.

Example uses: [Google Maps, QGIS, and Palladio maps](https://github.com/melaniewalsh/geospatial-lab) from a WUSTL digital humanities graduate seminar that I taught

Preview dataset

In [75]:
pd.read_csv("../data/nobel-prize-winners/nobel-prize-winners.csv").sample(10)

name        born  \
899                                       Pablo Neruda  1904-07-12   
450                          William Bradford Shockley  1910-02-13   
790                                      Ludwig Quidde  1858-03-23   
741                                Rabindranath Tagore  1861-05-07   
846                                      Henri Bergson  1859-10-18   
20                                        James Franck  1882-08-26   
763                                      Wolfgang Paul  1913-08-10   
926  Count Maurice (Mooris) Polidore Marie Bernhard...  1862-08-29   
773                                 Ernst Otto Fischer  1918-11-10   
296                                        Ferid Murad  1936-09-14   

           died bornCountry_original bornCountry_now bornCity_original  \
899  1973-09-23                Chile           Chile            Parral   
450  1989-08-12       United Kingdom  United Kingdom            London   
790  1941-03-04              Germany         Germany            Bremen   
741  1941-08-07                India           India          Calcutta   
846  1941-01-04               France          France             Paris   
20   1964-05-21              Germany         Germany           Hamburg   
763  1993-12-07              Germany         Germany       Lorenzkirch   
926  1949-05-06              Belgium         Belgium             Ghent   
773  2007-07-23              Germany         Germany            Munich   
296  0000-00-00                  USA             USA       Whiting, IN   

    bornCity_now            bornLocation   bornLong    bornLat  ...  \
899       Parral           Parral, Chile -71.822753 -36.140648  ...   
450       London  London, United Kingdom  -0.127758  51.507351  ...   
790       Bremen         Bremen, Germany   8.801694  53.079296  ...   
741     Calcutta         Calcutta, India  88.363895  22.572646  ...   
846        Paris           Paris, France   2.352222  48.856614  ...   
20       Hamburg        Hamburg, Germany   9.993682  53.551085  ...   
763  Lorenzkirch    Lorenzkirch, Germany  13.243844  51.356003  ...   
926        Ghent          Ghent, Belgium   3.717424  51.054342  ...   
773       Munich         Munich, Germany  11.581980  48.135125  ...   
296  Whiting, IN        Whiting, IN, USA -87.494487  41.679758  ...   

       diedLong    diedLat           diedCoordinates gender  year    category  \
899  -70.669266 -33.448890  -33.4488897, -70.6692655   male  1971  literature   
450 -122.143020  37.441883  37.4418834, -122.1430195   male  1956     physics   
790    6.143158  46.204391     46.2043907, 6.1431577   male  1927       peace   
741   88.363895  22.572646      22.572646, 88.363895   male  1913  literature   
846    2.352222  48.856614      48.856614, 2.3522219   male  1927  literature   
20     9.915804  51.541280     51.5412804, 9.9158035   male  1925     physics   
763    7.098207  50.737430       50.73743, 7.0982068   male  1989     physics   
926    7.261953  43.710173     43.7101728, 7.2619532   male  1911  literature   
773   11.581980  48.135125    48.1351253, 11.5819805   male  1973   chemistry   
296         NaN        NaN                       NaN   male  1998    medicine   

                                            motivation  \
899  "for a poetry that with the action of an eleme...   
450  "for their researches on semiconductors and th...   
790                                                NaN   
741  "because of his profoundly sensitive, fresh an...   
846  "in recognition of his rich and vitalizing ide...   
20   "for their discovery of the laws governing the...   
763    "for the development of the ion trap technique"   
926  "in appreciation of his many-sided literary ac...   
773  "for their pioneering work, performed independ...   
296  "for their discoveries concerning nitric oxide...   

                                       institutionName    institutionCity  \
899                                                NaN                NaN 

---

### Refugee Arrivals to the U.S.
(2005-2015)

Get the data: {download}`Download U.S. Refugee Arrivals data <../data/us-refugee-arrivals.zip>`  
Original source: Department of State's Refugee Processing Center and [Jeremy Singer-Vine](https://github.com/BuzzFeedNews/2015-11-refugees-in-the-united-states)

Brief description:   

The {download}`U.S. Refugee Arrivals zip file <../data/us-refugee-arrivals.zip>` contains data about refugee arrivals to the United States between 2005 and 2015. This data was originally compiled from the Department of State's Refugee Processing Center by Jeremy Singer-Vine for his BuzzFeed article ["Where U.S. Refugees Come From — And Go — In Charts."](https://www.buzzfeednews.com/article/jsvine/where-us-refugees-come-from-and-go-in-charts#.vooNwy74jO)

The "refugee-arrivals-by-destination" csv contains information about the number of refugees who arrived in each U.S. city and state, the year that they arrived, and the country from which they arrived. The "refugee-arrivals-by-religion" csv contains information about the number of refugees who arrived in the U.S., the year in which they arrived, and their religious affiliation.

Example uses:
- ["Where U.S. Refugees Come From — And Go"](https://www.buzzfeednews.com/article/jsvine/where-us-refugees-come-from-and-go-in-charts#.vooNwy74jO), Jeremy Singer-Vine
- [Tableau map](https://public.tableau.com/profile/melanie.walsh#!/vizhome/RefugeeArrivalstotheU_S_2005-2015/TotalRefugeeArrivalstoU_S_2005-2015) from a WUSTL digital humanities graduate seminar that I taught

Preview dataset

In [76]:
pd.read_csv("../data/us-refugee-arrivals/refugee-arrivals-by-destination.csv").sample(10)

year                  origin     dest_state     dest_city  arrivals
23778   2012  Bosnia and Herzegovina        Florida       Orlando         0
118184  2005                   Sudan     Washington   Federal Way         2
73846   2008             Afghanistan       New York         Bronx         0
80869   2013                   Yemen       New York      New York         0
92126   2006                   China   Pennsylvania  Philadelphia         0
114051  2008                  Bhutan     Washington    Des Moines         0
32673   2008                 Nigeria        Georgia       Decatur        15
98287   2007                Colombia      Tennessee     Nashville         0
51835   2008                   Burma  Massachusetts     Lexington         0
38155   2012                    Iraq       Illinois   Tinley Park         0

---

### Irish Immigrants Admitted to NYC's Bellevue Almshouse
(1840s)

Get data: [Link to Bellevue Almshouse data](https://docs.google.com/spreadsheets/d/1uf8uaqicknrn0a6STWrVfVMScQQMtzYf5I_QyhB9r7I/edit#gid=2057113261)  
Source: Anelise Shrout, [Digital Almshouse Project](https://www.nyuirish.net/almshouse/)

Brief description:   

The [Bellevue Almshouse link](https://docs.google.com/spreadsheets/d/1uf8uaqicknrn0a6STWrVfVMScQQMtzYf5I_QyhB9r7I/edit#gid=2057113261) takes you to a Google spreadsheet that contains data about Irish-born immigrants who were admitted to the Bellevue Almshouse in the 1840s. The Bellevue Almshouse was part of New York City's public health system, a place where poor, sick, homeless, and otherwise marginalized people were sent — sometimes voluntarily and sometimes forcibly. This dataset was transcribed from the almshouse's own admissions records by Anelise Shrout. For more information about this dataset, see [The Almshouse Records](https://www.nyuirish.net/almshouse/the-almshouse-records/)

Example uses:
- <a href=http://crdh.rrchnm.org/essays/v01-10-(re)-humanizing-data/>“(Re)Humanizing Data: Digitally Navigating the Bellevue Almshouse”</a>, Anelise Hanson Shrout
- ["Pandas Basics Part 1"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Data-Analysis/Pandas-Basics-Part1.html), *Introduction to Cultural Analytics & Python*

Preview dataset

In [42]:
pd.read_csv("../data/bellevue_almshouse_modified.csv").sample(5)

date_in first_name  last_name   age   disease profession gender  \
6247  1847-08-05      Jerry    Donohue  30.0  sickness    laborer      m   
9406  1846-11-09      James     Wilson   4.0       NaN    laborer      m   
7489  1847-11-23     Thomas       Wood  38.0  sickness    laborer      m   
8361  1847-06-01       Rose       Hall  32.0  sickness    married      w   
7933  1847-07-08       Mary  Creighton  62.0  sickness      widow      w   

     children  
6247      NaN  
9406      NaN  
7489      NaN  
8361      NaN  
7933      NaN

---

## Social Media 🕸️


### Donald Trump's Tweets
(2009-2020)

Get the data: {download}`Download Donald Trump tweets data <../texts/politics/Trump-Tweets.csv>`  
Original Source: [Trump Twitter Archive](http://www.trumptwitterarchive.com/)

Brief description:   

The {download}`Donald Trump tweets CSV file (.csv) <../texts/politics/Trump-Tweets.csv>` contains nearly 30,000 tweets from Donald Trump's account from 2009 to March 2020. The information about each tweet includes the source, tweet text, date of tweet, as well as retweet and favorite counts. More updated data can be downloaded at [Trump Twitter Archive](http://www.trumptwitterarchive.com/archive).
 
Example uses:
- ["Topic Modeling Time Series"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Text-Analysis/Topic-Modeling-Time-Series.html), *Introduction to Cultural Analytics & Python*'s
- ["How Trump Reshaped the Presidency in 11,000 Tweets"](https://www.nytimes.com/interactive/2019/11/02/us/politics/trump-twitter-presidency.html), *New York Times*

Preview dataset

In [43]:
pd.read_csv("../texts/social-media/Trump-Tweets.csv").sample(5)

source                                               text  \
14083  Twitter for Android  Thank you to Chris Cox and Bikers for Trump - ...   
10644   Twitter for iPhone  Thank you to the great men and women of the Un...   
28968   Twitter Web Client  You can listen to my interview today with Jay ...   
26510   Twitter Web Client  Fans shouldn't worry.  We have adjusted the fi...   
22184  Twitter for Android               @FidanizzBig   You need both thanks.   

                created_at  retweet_count  favorite_count is_retweet  \
14083  07-17-2016 03:15:22           6339           20895      False   
10644  02-23-2018 22:46:47          15608           87142      False   
28968  08-19-2011 15:04:24             25               6      False   
26510  10-30-2012 19:07:22            154              31      False   
22184  09-08-2013 03:21:35              2               1      False   

                   id_str  
14083  754514805264974000  
10644  967168890232082000  
28968  104569407761691000  
26510  263356437315588000  
22184  376545828191871000

---

### "Am I The Asshole?" Reddit Posts

Get the data: {download}`Download "Am I The Asshole?" Reddit data <../data/top-reddit-aita-posts.csv>`

Brief description:  

The {download}`Am I The Asshole?" Reddit posts CSV file (.csv) <../data/top-reddit-aita-posts.csv>` contains 2,932 Reddits posts from the subreddit "Am I the Asshole?" that have at least an upvote score of 2,000. The information in the dataset includes the date of the post, title, body text, url, upvote score, number of comments, and number of crossposts. This data was collected with [PSAW](https://github.com/dmarx/psaw), a wrapper for the [Pushshift API](https://github.com/pushshift/api).

Example uses:
- ["Topic Modeling CSV Files"](https://melaniewalsh.github.io/Intro-Cultural-Analytics/Text-Analysis/Topic-Modeling-CSV.html), *Introduction to Cultural Analytics & Python*

Preview dataset

In [44]:
pd.read_csv("../data/top-reddit-aita-posts.csv").sample(5)

author                  full_date        date  \
1887       carmelacorleone  2019-07-02 20:54:36+00:00  2019-07-02   
2086        Maximum_Silver  2019-06-10 23:35:03+00:00  2019-06-10   
2197           GeneralJen8  2019-05-31 16:28:55+00:00  2019-05-31   
2718        CrazyBirboLady  2019-03-29 21:55:46+00:00  2019-03-29   
560   Concerned_boyfriend7  2019-10-22 20:04:20+00:00  2019-10-22   

                                                  title  \
1887                  AITA for cussing at my racist MIL   
2086  WIBTA for forbidding my Father in law from see...   
2197                   AITA for shouting at protesters?   
2718  AITA for telling a “friend” that at least “my ...   
560   AITA for purposely scaring my girlfriend when ...   

                                               selftext  \
1887  Hello. I am a 24F engaged to be married. My fi...   
2086  My father in law is the king of passive aggres...   
2197  My (f26) mum died of cancer when I was 15. For...   
2718  I’m a “small” chested woman. Not even that sma...   
560   My (20m) girlfriend (21f) commutes to and from...   

                                                    url      subreddit  \
1887  https://www.reddit.com/r/AmItheAsshole/comment...  AmItheAsshole   
2086  https://www.reddit.com/r/AmItheAsshole/comment...  AmItheAsshole   
2197  https://www.reddit.com/r/AmItheAsshole/comment...  AmItheAsshole   
2718  https://www.reddit.com/r/AmItheAsshole/comment...  AmItheAsshole   
560   https://www.reddit.com/r/AmItheAsshole/comment...  AmItheAsshole   

      upvote_score  num_comments  num_crossposts  
1887          2581           654               0  
2086         16249          1970               0  
2197          2924           636               0  
2718         16049          2641               0  
560           2018           799               0

---

## Food 🍔


### The New York Public Library's Menu Dataset
(1840-present)

Get the data: [Link to The New York Public Library's Menu Dataset](http://menus.nypl.org/data)  
Source: The New York Public Library

Brief description:   

The [The New York Public Library's menu dataset link](http://menus.nypl.org/data) takes you to a web page where you can download data from the New York Public Library's massive menu collection — tens of thousands of transcribed menus and menu items from the 1840s to the present. Click "Download the latest data export in CSV format" for the most updated menu data.

Example uses:
- [*Curating Menus*](http://curatingmenus.org/), Katie Rawson and Trevor Muñoz

Preview dataset

In [50]:
pd.read_csv("../data/NYPL_Menus_2020_11_01_07/Dish.csv").sample(5)

id                              name  description  menus_appeared  \
370442  456603             Sliced Cucumber salad          NaN               2   
88857   111607      Quail Pie, New England Style          NaN               1   
37386    48141                     Grape Sherbet          NaN               7   
212711  270784  Pressed Lamb's Tongue, Chow Chow          NaN               1   
176554  223557                 Sundaes Chocolate          NaN               1   

        times_appeared  first_appeared  last_appeared  lowest_price  \
370442               2               0              0          0.00   
88857                1            1899           1899          0.00   
37386                9            1901           1973          0.25   
212711               1            1961           1961          0.00   
176554               1            1965           1965          0.75   

        highest_price  
370442           0.00  
88857            0.00  
37386            0.25  
212711           0.00  
176554           0.75

---

## Other Dataset Compilations

Below are some other great compilations of cultural and humanities-related datasets:
- Jeremy Singer-Vine's [*Data Is Plural* archive](https://docs.google.com/spreadsheets/d/1wZhPLMCHKJvwOkP4juclhjFgqIY8fQFMemwKL2c64vk/edit#gid=0) (you can [subscribe to his excellent dataset newsletter here](https://tinyletter.com/data-is-plural))
- *The Pudding*'s [GitHub repository](https://github.com/the-pudding/data)
- Alan Liu's [DH Toychest](http://dhresourcesforprojectbuilding.pbworks.com/w/page/69244469/Data%20Collections%20and%20Datasets)
- Reddit's [r/datasets subreddit](https://www.reddit.com/r/datasets/)

## Suggestions?

If you'd like to add a dataset or an example use case, please open [an issue on GitHub](https://github.com/melaniewalsh/Intro-Cultural-Analytics/issues) or email me at melanie.walsh@cornell.edu